src/flag_gems/ops/cummin.py

```diff
@@ -0,0 +1,341 @@
+import logging
+import math
+
+import torch
+import triton
+import triton.language as tl
+import triton.language.core as core
+
+from ..utils import libentry
+
+
+@triton.jit
+def _min_combine(a, idxa, b, idxb):
+    if a < b:
+        return a, idxa
+    else:
+        return b, idxb
+
+
+@triton.jit
+def tl_cummin(input, index, axis=0):
+    return core.associative_scan((input, index), axis, _min_combine)
+
+
+@triton.jit
+def _argmin_combine_tie_break_right(value1, index1, value2, index2):
+    tie = value1 == value2 and index1 > index2
+    lt = value1 < value2 or tie
+    value_ret = core.where(lt, value1, value2)
+    index_ret = core.where(lt, index1, index2)
+    return value_ret, index_ret
+
+
+@triton.jit
+def tl_min_tie_break_right(input, axis=None, keep_dims=False):
+    return core._reduce_with_indices(
+        input, axis, _argmin_combine_tie_break_right, keep_dims=keep_dims
+    )
+
+
+@libentry()
+@triton.jit(do_not_specialize=["n_elements"])
+def add_base_min_kernel(
+    out,
+    out_indices,
+    partial_min,
+    partial_min_indices,
+    n_elements,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    offset = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = offset < n_elements
+
+    out_ptrs = out + offset
+    out_indices_ptrs = out_indices + offset
+    out_vals = tl.load(out_ptrs, mask=mask)
+    out_indices = tl.load(out_indices_ptrs, mask=mask)
+
+    if pid > 0:
+        partial_min_ptrs = partial_min + pid - 1
+        last_part_min_via_min = tl.load(partial_min_ptrs)
+        partial_min_indices_ptrs = partial_min_indices + pid - 1
+        last_part_min_index_via_min = tl.load(partial_min_indices_ptrs)
+
+        final_vals = tl.minimum(out_vals, last_part_min_via_min)
+        final_indices = tl.where(
+            out_vals <= last_part_min_via_min, out_indices, last_part_min_index_via_min
+        )
+        tl.store(out_ptrs, final_vals.to(out_vals.dtype), mask=mask)
+        tl.store(out_indices_ptrs, final_indices, mask=mask)
+
+
+@libentry()
+@triton.jit(do_not_specialize=["n_elements"])
+def scan_part_min_kernel(
+    inp,
+    out,
+    out_indices,
+    partial_min,
+    partial_min_indices,
+    n_elements,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    offset = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = offset < n_elements
+
+    inp_ptrs = inp + offset
+    inp_vals = tl.load(inp_ptrs, mask=mask)
```

Comment:
Since this kernel for cummin, I think inp_vales should be loaded with padding value `inf` to avoid wrong result.

Reply:
Got it!